### PR TITLE
Jetpack: properly display plans with no expiration date

### PR DESCRIPTION
**Pull request by sergeymitr**
## Proposed changes:
* My Plan: Adjust styling for plans with no expiration date (Golden Token, Pressable Premium)

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
p8oabR-1aA-p2#comment-7299
p1683321216497689-slack-C051H2TQR54
8267-gh-Automattic/jpop-issues

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect a Jetpack site on a non-Golden Token plan
2. Go to "Jetpack Dashboard -> My Plan" and confirm that the "My Plan" card looks good.
3. Use the "Jetpack Golden Tokens" tool to get your site a Golden Token. Confirm that the card looks in accordance with the design (see the Slack conversation).

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/components/product-activated/index.jsx [hunk 1]
```diff
@@ -12,9 +12,15 @@ const ProductActivated = ( { type } ) => {
...
+		<div
+			className={
+				'jp-product-activated-label' + ( type === 'never-expires' ? ' never-expires' : '' )
+			}
+		>
 			<span className="jp-product-activated-label__text">
-				{ __( 'Subscription active', 'jetpack' ) }
+				{ type === 'never-expires'
+					? __( 'Never expires', 'jetpack' )
+					: __( 'Subscription active', 'jetpack' ) }
```
Comment (samiff) [inline projects/plugins/jetpack/_inc/client/components/product-activated/index.jsx]: Looks like the build may have tripped up on [this](https://github.com/Automattic/jetpack/actions/runs/4918503296/jobs/8784922187?pr=30519#step:8:8456).

May need to look at: https://www.npmjs.com/package/@automattic/i18n-check-webpack-plugin#user-content-conditional-function-call-compaction

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js [hunk 2]
```diff
@@ -72,7 +72,12 @@ class MyPlanHeader extends React.Component {
 				if ( ! isInTheFuture( purchase.expiry_date ) && purchase.expiry_date !== null ) {
 					activation = <ProductActivated key="product-expired" type="product-expired" />;
 				} else {
-					activation = <ProductActivated key="product-activated" />;
+					activation = (
+						<ProductActivated
+							key="product-activated"
+							type={ purchase.expiry_date === null ? 'never-expires' : '' }
```
Comment (samiff) [inline projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js:78]: Looks like there is a similar display issue reported here: 8267-gh-Automattic/jpop-issues

I _think_ this PR should handle that null expiry_date as well, but haven't tested as the PR build failed. Feel free to tag me for a review once this one is ready 👍

File: projects/plugins/jetpack/_inc/client/components/product-activated/index.jsx (hunk 1 above)
Comment (sergeymitr) [reply projects/plugins/jetpack/_inc/client/components/product-activated/index.jsx]: Fixed, thanks!

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js (hunk 2 above)
Comment (sergeymitr) [reply projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js:78]: Thanks for pointing out. I wasn't able to reproduce it, but will keep digging.

Comment (samiff) [reply projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js:78]: I was able to see the issue with one of my old test sites, and looks like this PR has helped.

Before:

Screenshot 2023-05-08 at 3 18 18 PM

After:

Screenshot 2023-05-08 at 3 18 58 PM

**Comment by jeffgolenski** [comment]
@sergeymitr Looks great, as always Sergey. 

Screen Shot 2023-05-09 at 10 42 37 AM


I just have a request for adjusting the mobile screens (for all plans). Feel free to ship this PR and create a new one for these changes if that's better for you.

**Currently**

On small screens the plan icon and text fight for space:

Screen Shot 2023-05-09 at 10 44 46 AM


**Proposed**

On small screens can we move the icon to the right and give the icon a left margin of 16px (right margin 0), just so the text stays flush to the left?

Screen Shot 2023-05-09 at 10 46 50 AM

**Review by jeherve — COMMENTED** [review]
This is what I see for my Golden ticket on `187771256 :

image

Shouldn't I see the Golden logo too?

**Comment by zinigor** [comment]
@sergeymitr can you please take a look at Jeremy's comment? I'm going to remove the Needs Review label for now.

**Comment by jeffgolenski** [comment]
@sergeymitr Per the mobile changes, we're almost there. If we can just make two adjustments:

1) Align the icon flush with the top of the headline (See screenshot) on mobile sizes. 24px padding on the top seems to do it!
2) On all screen sizes, can we adjust the line-height of p.my-plan-card__tag-line from 18px to 1.65?

Current
Frame 12650


Proposed
Screen Shot 2023-05-11 at 1 55 05 PM

**Comment by sergeymitr** [comment]
Hi @zinigor, thanks for removing the label.
Jeremy's comment addressed here: p1683767271692959-slack-C051H2TQR54
I need to adjust the styling per last Jeff's comment and we should be good for another round of reviews :)

**Review by samiff — APPROVED** [review]
Code looks good and tests well on my end :shipit:

**Pull request merged**